### PR TITLE
querier: Added select benchmark

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

Needed for future measurements. cc @spaparaju @ianbillett

### CODE REVIEW

**Comment by bwplotka** [comment]
PTAL @GiedriusS @onprem @squat @spaparaju  - I am kind of blocked on this to work on further querier optimizations. 🤗

**Review by onprem — APPROVED** [review]
I have low context about Querier internals. Keeping that in mind, this looks good to me.

File: pkg/query/query_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		// Note 0 argument - this is because we want to have two replicas for the same time duration.
+		head, created := storetestutil.CreateHeadWithSeries(t, 0, storetestutil.HeadGenOptions{
+			TSDBDir:          filepath.Join(tmpDir, fmt.Sprintf("%d", j)),
+			SamplesPerSeries: samplesPerSeriesPerReplica,
+			Series:           seriesPerReplica,
+			Random:           random,
+			PrependLabels:    labels.FromStrings("a_replica", fmt.Sprintf("%d", j)), // a_ prefix so we keep sorted order.
+		})
+		testutil.Ok(t, head.Close())
+		for i := 0; i < len(created); i++ {
```
Comment (ianbillett) [inline pkg/query/query_bench_test.go:70]: This whole section is difficult parse what is going on and why - care to add a comment?

**Review by ianbillett — APPROVED** [review]
Everything runs locally for me :+1: 

Minor comments on readability / understandability

File: pkg/query/query_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/thanos-io/thanos/pkg/gate"
+	"github.com/thanos-io/thanos/pkg/store/labelpb"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	storetestutil "github.com/thanos-io/thanos/pkg/store/storepb/testutil"
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+// TestQuerySelect benchmarks querier Select method. Note that this is what PromQL is using, but PromQL might invoke
```
Comment (ianbillett) [inline pkg/query/query_bench_test.go:24]: Surely `TestQuerySelect` tests the querier select method, and `BenchmarkQuerySelect` benchmarks the querier select method?

**Comment by bwplotka** [comment]
Thanks, for review, I will fix this in next PR!

**Pull request merged**